### PR TITLE
Test improvements

### PR DESCRIPTION
**Pull request by kripken**
Remove legacy asm2nn test mode (non-native optimizer, no longer important) and add travis binaryen tests.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,6 +18,12 @@ env:
 - TEST_TARGET="asm2nn.test_p* asm3.test_p*"
```
[the remark this answers is not in the document]
Comment (kripken) [reply .travis.yml]: Thanks! I missed that, heh...

**Comment by kripken** [comment]
Hmm, a bigger issue is that these wasm tests need a js engine with wasm, and we use node 4.1.1 which does not have it. So we'd need to get and install one.

For now, I made this PR just remove the `asm2nn` legacy test. Leaving figuring out wasm testing for later.

**Comment by juj** [comment]
I have a TODO item on my table to update emsdk to latest node.js, would that help here?

**Comment by curiousdannii** [comment]
Looks like a few more test jobs could be combined, and every time you do you cut 4 minutes off the running time, and probably more from not needing to wait for a VM to be assigned.

ALL.test_a* and ALL.test_b*/ALL.test_c* (depending on how happy you are for them to get out of order)
asm3.test_d* and ALL.test_e*
asm2i.test_p* and asm3.test_p*

**Comment by SaschaNaz** [comment]
Today I tried Circle CI to get better performance and I found that it can finish `TEST_TARGET=other` in 3m 20s (compared to 40m in Travis 😲).

The problem is it's not really free, with 1500 build minutes/month limit for the free plan.

**Comment by kripken** [comment]
> I have a TODO item on my table to update emsdk to latest node.js, would that help here?

I'm not sure latest stable would be enough. We keep supporting new wasm features as they come. I think we'd eventually want to grab nightly builds of js vms or something like that.

**Comment by kripken** [comment]
> Today I tried Circle CI to get better performance and I found that it can finish TEST_TARGET=other in 3m 20s (compared to 40m in Travis 😲). The problem is it's not really free, with 1500 build minutes/month limit for the free plan.

Hmm :) Sounds amazing, but also too good to be true, even on my fast local machines it's much closer to 40m than 3m. Does the log really show it completed all the tests in that time?

**Comment by SaschaNaz** [comment]
Yes, you can see the logs here: https://circleci.com/gh/SaschaNaz/emscripten/99 (though it took 7m this time because of EMCC_CORES=4)

**Comment by SaschaNaz** [comment]
Oh sorry, I found an error on my comment and fixed it. You can still see the logs here:

Best performance I've seen, with default EMCC_CORES=32 (but failing): https://circleci.com/workflow-run/9e9acbb3-16c3-4dce-8a47-96395b9c40f8

All green but slower with EMCC_CORES=4: https://circleci.com/workflow-run/f48cf61d-e8ef-4537-9092-c20d4fc4631a

**Comment by SaschaNaz** [comment]
Sorry, I've found an error on my previous comment and fixed it.

You can track logs here: https://circleci.com/gh/SaschaNaz/workflows/emscripten/tree/circleci

(Direct link to a specific workflow requires login 😅, sorry for repeated comments+deletions)

**Comment by kripken** [comment]
Still very fast, impressive. I guess circle-ci bots have many cores? Seems like an interesting option.

**Comment by kripken** [comment]
Merging this PR, which just removes the obsolete `asm2nn` mode. Further test improvements left for later.

**Comment by SaschaNaz** [comment]
>I guess circle-ci bots have many cores?

The assigned resources are 2CPUs+4096MB RAM which are not so massive, not sure what causes the vast performance difference. Maybe the current docker-in-docker on Travis causes performance degradation?

**Comment by curiousdannii** [comment]
>[What if I am building open-source?](https://circleci.com/pricing/#faq-section-linux)

>We offer a total of four free linux containers ($2400 annual value) for open-source projects. Simply keeping your project public will enable this for you!

That might be enough?

**Comment by SaschaNaz** [comment]
That doesn't say anything about what happens on the "1,500 build minutes per month" limit, though.

**Comment by curiousdannii** [comment]
Probably worth kripken emailing them to see what they can offer.

**Comment by kripken** [comment]
For now let's see how we do with the current setup. The speed has been fine I think. If it becomes a problem we can investigate the other options.

**Pull request merged**